mongo/options/clientoptions.go

```diff
@@ -728,8 +729,9 @@ func (c *ClientOptions) SetSocketTimeout(d time.Duration) *ClientOptions {
 // be honored if there is no deadline on the operation Context. Timeout can also be set through the "timeoutMS" URI option
 // (e.g. "timeoutMS=1000"). The default value is nil, meaning operations do not inherit a timeout from the Client.
 //
-// If any Timeout is set (even 0) on the Client, the values of other, deprecated timeout-related options will be ignored.
-// In particular: ClientOptions.SocketTimeout, WriteConcern.wTimeout, MaxTime on operations, and TransactionOptions.MaxCommitTime.
+// If any Timeout is set (even 0) on the Client, the values of MaxTime on operation options and TransactionOptions.MaxCommitTime
+// will be ignored. Setting ClientOptions.SocketTimeout or WriteConcern.wTimeout when a Timeout is set on the Client will result
+// in undefined behavior.
```

Comment:
I've tried to update the documentation to reflect the fact that when `Timeout` is set: `MaxTime` and `MaxCommitTime` are ignored, and `SocketTimeout` and `wTimeout` can create undefined behavior (but are not ignored).